app/controllers/dashboards_controller.rb

```diff
@@ -3,7 +3,6 @@ class DashboardsController < BaseController
   before_action :set_policy_with_context, except: [:refresh_dashboard]
   before_action :set_dashboard
   before_action :set_defence_requests, only: [:show, :closed]
-  before_action :set_dashboard_count, only: [:show, :closed]
```

Comment:
Sure. So in `routes.rb`: 

```ruby
  resources :dashboard, only: :show
```

Then in the `dashboards_controller.rb`:

```ruby
  def show
    type = params[:id]
    set_defence_requests(type)
    set_dashboard_count(type)
  end
```

These methods then would check that the type is allowed and use the appropriate scopes on `DefenceRequest`. 

I hope that makes sense :)